src/main/java/seedu/module/model/module/exceptions/LinkCreationException.java

```diff
@@ -0,0 +1,10 @@
+package seedu.module.model.module.exceptions;
+
+/**
+ * Signals that the url provided by user end is not a proper link recognized by program
+ */
+public class LinkCreationException extends RuntimeException {
```

Comment:
Why must this be a runtime exception? Why not just a normal exception?

Reply:
i thought it would be best if i follow the rest of the exceptions contained under the module/exceptions folder.

Reply:
Tracing through the code, I realised that the other runtime exceptions are never actually properly used (i.e. dead code).

I would really like to avoid unchecked exceptions unless necessary.

[According to this thread](https://stackoverflow.com/questions/27578/when-to-choose-checked-and-unchecked-exceptions), a checked exception should be used for predictable but unpreventable errors that are reasonable to recover from. In this case, we can predict that a user may enter an invalid url name and recover from it by telling the user that it is invalid, hence we should use a checked exception.